Gateway/Command/Cancel.php

```diff
@@ -104,10 +104,12 @@ public function execute(array $commandSubject)
         $transactionObject = new TransactionObject();
         $transactionObject->setTransaction($transaction);
 
-        $cancelRequest = $this->getRequestService('Paymentorder/Transaction', 'TransactionCancel', $transactionObject);
-        $cancelRequest->setRequestEndpointVars(
-            $this->getSwedbankPayPaymentResourceId($paymentOrder->getPaymentOrderId())
+        $cancelRequest = $this->getRequestService(
+            'Paymentorder/Transaction',
+            'TransactionCancel',
+            $transactionObject
         );
+        $cancelRequest->setPaymentOrderId($paymentOrder->getPaymentIdPath());
```

Comment:
Why does `setPaymentOrderId` take `paymentIdPath` here? Is the `setPaymentOrderId` method misnamed?

```suggestion
        $cancelRequest->setPaymentId($paymentOrder->getPaymentId());
```

Reply:
As you know, since the initial solution, we had been parsing the `payment_order_id` from the response to extract the last part and we would use it to process different requests. However, based on the suggestions from you, we are now using the entire ID and saving it as `paymentIdPath` just to avoid confusions. 

So the above code is correct here.